File: 26 -- OutaTimeOC/OutaTime/OutaTime/DatePickerViewController.m
//
//  DatePickerViewController.m
//  OutaTime
//
//  Created by Jennifer Hamilton on 11/9/15.
//  Copyright © 2015 The Iron Yard. All rights reserved.
//

#import "DatePickerViewController.h"

@interface DatePickerViewController ()

@end

@implementation DatePickerViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

- (IBAction)datePicker:(UIDatePicker *)sender
{
//    delegate?.destinationDateWasChosen(picker.date)    
    [self.delegate dateWasPicked: sender.date];
}
@end

File: 04 -- OutaTime/OutaTime/OutaTime/DatePickerViewController.swift
//
//  DatePickerViewController.swift
//  OutaTime
//
//  Created by Jennifer Hamilton on 10/8/15.
//  Copyright © 2015 The Iron Yard. All rights reserved.
//

import UIKit


class DatePickerViewController: UIViewController
{
    @IBOutlet var picker: UIDatePicker!
    // delegate property here. this is the 'sending' class
    var delegate: DatePickerDelegate? //question mark means 'optional' (doesnt' have to have a value)
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func viewWillDisappear(animated: Bool)
    {
        super.viewWillDisappear(animated)
        // '?' means only run if delegate has a value
        delegate?.destinationDateWasChosen(picker.date)/*this should be data sent to timecircuits*/
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}